sdk/dotnet/Pulumi/Core/Output.cs

```diff
@@ -58,7 +58,13 @@ public static Output<ImmutableArray<T>> All<T>(params Input<T>[] inputs)
         /// </summary>
         public static Output<ImmutableArray<T>> All<T>(IEnumerable<Input<T>> inputs)
             => Output<T>.All(ImmutableArray.CreateRange(inputs));
-        
+
+        /// <summary>
+        /// <see cref="All{T}(Input{T}[])"/> for more details.
+        /// </summary>
+        public static Output<ImmutableArray<T>> All<T>(ImmutableArray<Input<T>> inputs)
+            => Output<T>.All(ImmutableArray.CreateRange(inputs.EnsureNotDefault()));
+
```

Comment:
Actually maybe it might be better to remove the `ImmutableArray` overloads and update the `IEnumerable` ones such as:

```csharp
public static Output<ImmutableArray<T>> All<T>(IEnumerable<Input<T>> inputs)
    => Output<T>.All(ImmutableArray.CreateRange(inputs switch {
            ImmutableArray<Input<T>> immutable => immutable.EnsureNotDefault(),
            _ => inputs,
        }));

public static Output<ImmutableArray<T>> All<T>(IEnumerable<Output<T>> outputs)
    => Output<T>.All(ImmutableArray.CreateRange((outputs switch {
            ImmutableArray<Output<T>> immutable => immutable.EnsureNotDefault(),
            _ => outputs,
        }).Select(o => (Input<T>)o)));
```
Doing this would also ensure the following tests work

```csharp
await AssertEmpty(Output.All((IEnumerable<Input<bool>>)(ImmutableArray<Input<bool>>) default));
await AssertEmpty(Output.All((IEnumerable<Output<bool>>)(ImmutableArray<Output<bool>>) default));
```